app/services/idempotency_records/key_service.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module IdempotencyRecords
+  class KeyService < BaseService
+    Result = BaseResult[:idempotency_key]
+
+    # WARNING: changing this value is very dangerous!
+    # only do this if you really have to.
+    # Uniqueness for existing values can no longer be enforced once this is changed
+    KEY_VERSION = "v1"
+    SEPARATOR = "|"
+
+    def initialize(*key_parts)
+      @key_parts = key_parts
+
+      super()
+    end
+
+    def call
+      string_to_digest = key_parts.map(&:to_s).join(SEPARATOR)
+      result.idempotency_key = Digest::SHA256.digest("#{KEY_VERSION}#{SEPARATOR}#{string_to_digest}")
```

Comment:
I understand that we like services but I feel like this should really be on the model. The constant for version and separator especially.

Reply:
The model does not store the key parts though?

Reply:
@julienbourdeau I'm thinking this has a chance of causing some unexpected bugs (like imagine using booleans for these values, order then is suddenly really important). Do you see any downside to requiring that the order is also important?

Reply:
I didn't mean to store them but just to have the constant and a "build" method on the model. This is really a key feature of the model, usually services do stuff "around" the model.
This is not really important tho, it just felt weird on this one but feel free to ignore.

```
def build_key(key_parts)
  Digest::SHA256.digest("#{KEY_VERSION}#{SEPARATOR}#{key_parts.map(&:to_s).join(SEPARATOR)}")
end
```